client/python/samples/utils/vehicle_utils.py

```diff
@@ -27,7 +27,12 @@ def vehicle_postprocess(response, image_path, output_name, output_save_path):
     confidence_threshold = 0.5
     for i in range(output.shape[2]):
         image = draw_bounding_box(output[0, 0, i, ...], image, width, height, confidence_threshold)
-    cv2.imwrite(os.path.join(output_save_path, 'vehicle-detection' + '.jpg'), image)
+    if not os.path.exists(output_save_path):
+        print("Making directory for output: " + output_save_path)
+        os.makedirs(output_save_path)
+    output_file = os.path.join(output_save_path, img_index + '.jpg')
```

Comment:
It's nice we can now save multiple predictions, but working with index seems unnecessary. 
Could you try using the same name as for the input? There can't be two files with the same name in one directory so it would be safe and if user has images in paths:
```
/home/user/data/input_images/medium_cars.jpg
/home/user/data/input_images/small_cars.jpg
/home/user/data/input_images/trucks.jpg
```
Then providing output path like `/home/user/data/output_images` would save results as:
```
/home/user/data/output_images/medium_cars.jpg
/home/user/data/output_images/small_cars.jpg
/home/user/data/output_images/trucks.jpg
```

Simply extract the last part of the input path and append it to the output path.